packages/app-builder-lib/src/targets/nsis/NsisTarget.ts

```diff
@@ -319,7 +322,7 @@ export class NsisTarget extends Target {
       updateInfo = await createBlockmap(installerPath, this, packager, safeArtifactName)
     }
 
-    if (updateInfo != null && isPerMachine && oneClick) {
+    if ((updateInfo != null && isPerMachine && oneClick) || allowServiceToInstall) {
```

Comment:
Are you already using `packElevateHelper = true` by chance? Would this work instead?
```
(updateInfo != null && isPerMachine && (oneClick || options.packElevateHelper))
```

Reply:
Yes, elevate.exe file is alreay included in my electron app becauseo of perMachine true option. 

electron updater decides to elevate or not using isAdminRightsRequired in update info file.
(https://github.com/electron-userland/electron-builder/blob/5e381c556d12ce185bb7ea720380509c1ddc5cf7/packages/electron-updater/src/NsisUpdater.ts#:~:text=if%20(options.isAdminRightsRequired,%7D) So, it's seamless using packElevateHelper option beacuse if elevate.exe packed, then electron updater must using this to elevate when update.